src/dict.c

```diff
@@ -1452,7 +1452,7 @@ static void _dictExpandIfNeeded(dict *d)
     if ((dict_can_resize == DICT_RESIZE_ENABLE &&
          d->ht_used[0] >= DICTHT_SIZE(d->ht_size_exp[0])) ||
         (dict_can_resize != DICT_RESIZE_FORBID &&
-         d->ht_used[0] / DICTHT_SIZE(d->ht_size_exp[0]) > dict_force_resize_ratio))
+         (double)d->ht_used[0] / DICTHT_SIZE(d->ht_size_exp[0]) >= dict_force_resize_ratio))
```

Comment:
Hi, if we add an equal condition here, maybe we don't need to change the result of division to double because I think it's the same as `d->ht_used[0] / DICTHT_SIZE(d->ht_size_exp[0]) >= dict_force_resize_ratio`. WDYT?

Reply:
that is true, i just want to keep them consistent. if we don't want to go the double way, we can change it